src/Sarif.Viewer.VisualStudio/ErrorList/ErrorListService.cs

```diff
@@ -79,8 +79,15 @@ public static void ProcessLogFile(string filePath, Solution solution, string too
                 }
                 else
                 {
-                    // They're opening a v2 log, so send it through the pre-release compat transformer
-                    PrereleaseCompatibilityTransformer.UpdateToCurrentVersion(logText, Formatting.Indented, out logText);
+                    // They're opening a v2 log; check to see if it's the version we are referencing.
+                    pattern = $@"""version""\s*:\s*""{VersionConstants.StableSarifVersion}""";
```

Comment:
>pattern [](start = 20, length = 7)

1) I'm curious, maybe we should test whether this is in fact saving us any time. i.e., get an extremely large log file and test the version sniff vs. running it through the transformer.
2) one of the symptoms of that other empty log file was that I observed we seemed to be producing a temporary version of the file (I assumed because it had been transformed. I don't think this change will fix that because you aren't setting the output path. (i think you should do this below in the event that match.Success is true?)
3) we just updated the SDK on nuget.org, any chance we can sneak in an upgrade to it?

Reply:
btw - scott louvau can provide a SARIF log file with a large # of results for testing if you'd like to explore #1. I'm fine with deferring this work for now if you are busy. I think items #2 and #3 shouldn't cost much, so hopeful you can take them on. thanks a ton for this contribution, easy. 

---
In reply to: [287412015](https://github.com/microsoft/sarif-visualstudio-extension/pull/88#discussion_r287412015) [](ancestors = 287412015)

Reply:
OK thx will investigate.

Reply:
So what I did is:
1. Only run the regex against the first 100 characters. The spec says the version property should be first, so I think this is a fair assumption. A messagebox pops if the version property isn't found.
2. Fixed the temp file stuff. If the log is current, it's used in-place.
3. I was stupidly doing a second regex if the log wasn't v1. Refactored as part of item 1.
4. Updated the SDK package reference.

Reply:
5. Added a prompt to save the prerelease-transformed log, just like we do for v1->v2 transforms.